### PR TITLE
 Create organization & project dialog #LMR-984

### PR DESCRIPTION
**Pull request by kubedo8**

### CODE REVIEW

File: src/app/dialog/create-resource/create-resource-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  }
+
+  public ngOnDestroy() {
+    this.subscriptions.unsubscribe();
+  }
+
+  private createForm() {
+    this.form = new FormGroup({
+      code: new FormControl('', [minLengthValidator(2), maxLengthValidator(5)]),
+      name: new FormControl('')
```
[the remark this answers is not in the document]
Comment (kubedo8) [reply src/app/dialog/create-resource/create-resource-dialog.component.ts]: Name wasn't required so far. We can discuss this with @marvec

File: src/app/dialog/create-resource/create-resource-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    this.parseResourceType();
+  }
+
+  public ngOnDestroy() {
+    this.subscriptions.unsubscribe();
+  }
+
+  private createForm() {
+    this.form = new FormGroup({
+      code: new FormControl('', [minLengthValidator(2), maxLengthValidator(5)]),
```
[the remark this answers is not in the document]
Comment (kubedo8) [reply src/app/dialog/create-resource/create-resource-dialog.component.ts]: Build-in validators also takes account white spaces. I think we should always have trimmed value.

File: src/app/core/top-panel/top-panel.component.ts [hunk 3]
```diff
@@ -211,10 +217,24 @@ export class TopPanelComponent implements OnInit {
   }
 
   public createNewOrganization(): void {
-    // TODO call create dialog
+    this.dialogService.openCreateResourceDialog(ResourceType.Organization, null, this.onCreateOrganization());
```
[the remark this answers is not in the document]
Comment (kubedo8) [reply src/app/core/top-panel/top-panel.component.ts]: Oh thanks.

File: src/app/dialog/dialog-routing.module.ts [hunk 4]
```diff
@@ -51,6 +52,16 @@ const routes: Routes = [
...
   },
+  {
+    path: `${DialogPath.CREATE_RESOURCE}/:resourceType`,
+    component: CreateResourceDialogComponent,
+    outlet: 'dialog'
+  },
+  {
+    path: `${DialogPath.CREATE_RESOURCE}/:resourceType/:parentId`,
+    component: CreateResourceDialogComponent,
+    outlet: 'dialog'
```
[the remark this answers is not in the document]
Comment (kubedo8) [reply src/app/dialog/dialog-routing.module.ts]: Ok, that's good idea.

File: src/app/dialog/create-resource/create-resource-dialog.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (marvec) [reply src/app/dialog/create-resource/create-resource-dialog.component.ts]: This is correct - name is not required.

File: src/app/dialog/create-resource/create-resource-dialog.component.html [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        </icon-chooser>
+      </div>
+
+      <resource-code-input
+        [form]="form">
+      </resource-code-input>
+
+      <resource-name-input
+        [form]="form">
+      </resource-name-input>
```
[the remark this answers is not in the document]
Comment (marvec) [reply src/app/dialog/create-resource/create-resource-dialog.component.html]: Why don't we reuse input-box here? This is why we have it...

File: src/app/dialog/create-resource/create-resource-dialog.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (livthomas) [reply src/app/dialog/create-resource/create-resource-dialog.component.ts]: Why do we allow users to enter white space characters anyway?

File: src/app/dialog/create-resource/create-resource-dialog.component.ts [hunk 6]
```diff
@@ -143,24 +146,43 @@ export class CreateResourceDialogComponent implements OnInit, OnDestroy {
...
+    this.subscriptions.add(
+      this.store.select(selectProjectsByOrganizationId(this.parentId))
+        .subscribe(projects => this.isFirstProject = projects.length === 0)
+    );
+  }
+
+  private getResourceTypeFromRouter(): ResourceType {
+    const [rootPath,] = this.route.routeConfig.path.split('/');
+    const dialogPath = dialogPathsMap[rootPath];
+    console.log(rootPath, dialogPath);
```
Comment (livthomas) [inline src/app/dialog/create-resource/create-resource-dialog.component.ts]: `console.log`

**Pull request merged**